src/main/java/slimeknights/tconstruct/library/tools/capability/inventory/ToolInventoryCapability.java

```diff
@@ -274,7 +274,7 @@ public ItemStack insertItem(int slot, ItemStack stack, boolean simulate) {
     } else {
       // space leftover? does it match?
       int limit = Math.min(current.getMaxStackSize(), slotLimit);
-      if (current.getCount() >= limit || !current.sameItem(stack)) {
+      if (current.getCount() >= limit || !current.is(stack.getItem())) {
```

Comment:
Are you sure this is equivalent? The is method that takes an item is not new, IIRC this is supposed to be checking the item and NBT.

Reply:
Yep, because that's the underlying code of `isSameItem` in `ItemStack`.
~~~java
public static boolean isSameItem(ItemStack stack, ItemStack other) {
   return stack.is(other.getItem());
}
~~~

However, there is `isSameItemSameTags`, which also compares the NBT Tags. Maybe that's what we should use instead?
~~~java
public static boolean isSameItemSameTags(ItemStack pStack, ItemStack pOther) {
    if (!pStack.is(pOther.getItem())) {
        return false;
    } else {
        return pStack.isEmpty() && pOther.isEmpty() ? true : Objects.equals(pStack.tag, pOther.tag) && pStack.areCapsCompatible(pOther);
    }
}
~~~

Reply:
Note: the `sameItem()` method's source code is the following:

```java
   /**
    * Compares the {@code ItemStack} argument to this {@code ItemStack}, returns {@code true} if the {@code Items}
    * contained in both {@code ItemStacks} are equal.
    */
   public boolean sameItem(ItemStack pOther) {
      return !pOther.isEmpty() && this.is(pOther.getItem());
   }
```

So now, we're just calling `is()` directly, but skipping the check for `other.isEmpty()`, which is probably fine since we're already checking the count?

It's also worth noting that the new `isSameItem()` method that replaced `sameItem()` just calls `is()` without calling `isEmpty()`:

```java
    public static boolean isSameItem(ItemStack pStack, ItemStack pOther) {
        return pStack.is(pOther.getItem());
    }
```

Reply:
Just seems like an arbitrary change unless the method was removed. If this was a method removal that prompted it then its fine though it seems like something I would not expect to be removed.

As an aside, two empty stacks would have matching items due to the way `getItem()` works. So effectively the is empty check means that two empty stacks are not considered to have the same item despite both being empty.

Reply:
This change was made due to a method removal, yes -- `sameItem()` was replaced with `isSameItem()`, but since `isSameItem()` just calls `is()`, I opted to use `is()` directly!